src/controllers/product.controller.ts

```diff
@@ -0,0 +1,114 @@
+import { Context } from 'koa';
+import { model, Model } from "mongoose";
+import { productSchema } from "../schemas/product.schema";
+import { logger } from '../../config/winston/winston';
+import { Product } from '../models/product.model';
+import { validatePartialProduct, validateProduct } from '../validation/product.validation';
+
+const Product: Model<Product> = model<Product>('Product', productSchema);
+
+export async function getProducts(ctx: Context) {
+    try {
+        const products: Product[] = await Product.find();
```

Comment:
verify if this works well considering it is thenable and not promise

Reply:
Model.find() returns a query instance, so why can you do await Model.find()? That's because a Mongoose query is a thenable, meaning they have a then() function. That means you can use queries in the same way you use promises, including with promise chaining as shown below.